sql/src/main/java/org/apache/druid/sql/calcite/rule/DruidUnionRule.java

```diff
@@ -51,6 +52,13 @@ public DruidUnionRule(PlannerContext plannerContext)
   @Override
   public boolean matches(RelOptRuleCall call)
   {
+    if (!plannerContext.featureAvailable(EngineFeature.ALLOW_TOP_LEVEL_UNION_ALL)) {
```

Comment:
Better to simply not add the rule if this feature is missing. Try adding a conditional in `DruidRules.rules`. There are others there for other engine features.

Reply:
I was wondering if it's worth it for the cleaner exception message. I'll conditionally add the rule instead, since eventually, we'll be able to plan the query using union data source.